src/app/services/auth.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable, inject, signal } from '@angular/core';
+import {
+  Auth,
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+  signOut,
+  updateProfile,
+  user,
+} from '@angular/fire/auth';
+import { Observable, from } from 'rxjs';
+import { UserInterface } from '../models/user.interface';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  firebaseAuth = inject(Auth);
+  user$ = user(this.firebaseAuth);
+  currentUserSig = signal<UserInterface | null | undefined>(undefined);
```

Comment:
I don't actually see you setting this signal anywhere? unless I missed it, but just keep in mind that you do still have it here